core/client/fs/src/main/java/alluxio/client/file/cache/store/MemoryPageStore.java

```diff
@@ -29,15 +30,26 @@
  */
 @NotThreadSafe
 public class MemoryPageStore implements PageStore {
-  private ConcurrentHashMap<PageId, byte[]> mPageStoreMap = new ConcurrentHashMap<>();
+
+  private final PagePool mPagePool;
+
+  private ConcurrentHashMap<PageId, MemPage> mPageStoreMap = new ConcurrentHashMap<>();
+
+  /**
+   * Constructor of MemoryPageStore.
+   * @param pageSize page size
+   */
+  public MemoryPageStore(int pageSize) {
+    mPagePool = new PagePool(pageSize);
+  }
 
   @Override
   public void put(PageId pageId, ByteBuffer page, boolean isTemporary) throws IOException {
     //TODO(beinan): support temp page for memory page store
     PageId pageKey = getKeyFromPageId(pageId);
     try {
-      byte[] pageCopy = new byte[page.remaining()];
-      page.get(pageCopy);
+      MemPage pageCopy = mPagePool.getPage(page.remaining());
+      page.get(pageCopy.getPage(), 0, pageCopy.mPageLength);
```

Comment:
would it be possible that pageLength is larger than byte array length and causing exception?

Reply:
I don't think it's possible.  But we could put a checkState or checkArguments here to make it fail more early and explicitly.

Reply:
hmmm, I'm not so sure,  let me try to add more checkState in another pr.

Reply:
> would it be possible that pageLength is larger than byte array length and causing exception?

This is also my concern.